File: BringMeHome/BringMeHome.Services/Database/Animal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BringMeHome.Services.Database
{
    public class Animal
    {
        [Key]
        public int AnimalID { get; set; }

        [Required, StringLength(100)]
        public string Name { get; set; }

        public int BreedID { get; set; }

        public int Age { get; set; }

        [Required, StringLength(20)]
        public string Gender { get; set; }

        public decimal Weight { get; set; }

        [DataType(DataType.Date)]
        public DateTime DateArrived { get; set; }

        public int StatusID { get; set; }

        [StringLength(1000)]
        public string Description { get; set; }

        public byte[]? AnimalImage { get; set; }

        [StringLength(500)]
        public string HealthStatus { get; set; }

        public int ShelterID { get; set; }

        public int? ColorID { get; set; }

        public int? TempermentID { get; set; }

        // Navigation properties
        [ForeignKey("BreedID")]
        public virtual Breed Breed { get; set; }

        [ForeignKey("StatusID")]
        public virtual AnimalStatus Status { get; set; }

        [ForeignKey("ShelterID")]
        public virtual Shelter Shelter { get; set; }

        [ForeignKey("ColorID")]
        public virtual Color Color { get; set; }

        [ForeignKey("TempermentID")]
        public virtual AnimalTemperament AnimalTemperament { get; set; }

        public virtual ICollection<AdoptionApplication> AdoptionApplications { get; set; }
    }
}

File: bringmehomeUI/bringmehome_admin/lib/models/animal.dart
class Animal {
  int? animalID;
  String? name;
  String? description;
  int? breedID;
  String? breedName;
  int? age;
  String? gender;
  double? weight;
  DateTime? dateArrived;
  int? statusID;
  String? healthStatus;
  int? shelterID;
  String? shelterName;
  int? colorID;
  String? colorName;
  int? tempermentID;
  String? tempermentName;
  bool isFavorite;

  Animal({
    this.animalID,
    this.name,
    this.description,
    this.breedID,
    this.age,
    this.gender,
    this.weight,
    this.dateArrived,
    this.statusID,
    this.healthStatus,
    this.shelterID,
    this.colorID,
    this.colorName,
    this.tempermentID,
    this.tempermentName,
    this.breedName,
    this.shelterName,
    this.isFavorite = false,
  });

  factory Animal.fromJson(Map<String, dynamic> json) {
    return Animal(
      animalID: json['animalID'] as int?,
      name: json['name'] as String?,
      description: json['description'] as String?,
      breedID: json['breedID'] as int?,
      breedName: json['breedName'] as String?,
      age: json['age'] != null ? int.tryParse(json['age'].toString()) : null,
      gender: json['gender'] as String?,
      weight: json['weight'] != null
          ? double.tryParse(json['weight'].toString())
          : null,
      dateArrived: json['dateArrived'] != null
          ? DateTime.tryParse(json['dateArrived'].toString())
          : null,
      statusID: json['statusID'] as int?,
      healthStatus: json['healthStatus'] as String?,
      shelterID: json['shelterID'] as int?,
      shelterName: json['shelterName'] as String?,
      colorID: json['colorID'] as int?,
      colorName: json['colorName'] as String?,
      tempermentID: json['tempermentID'] as int?,
      tempermentName: json['tempermentName'] as String?,
      isFavorite: json['isFavorite'] as bool? ?? false,
    );
  }
}